discord/webhook.py

```diff
@@ -0,0 +1,119 @@
+from .embeds import Embed
+from .mixins import Hashable
+from .errors import HTTPException
+from .utils import _bytes_to_base64_data
+import aiohttp
+import io
+import json
+import asyncio
+
+# Webhook code made by Cerulean#7014 
+
+WEBHOOK_BASE = 'https://discordapp.com/api/webhooks/'
+AVATAR_BASE = 'https://cdn.discordapp.com/avatars/{id}/{avatar}.png'
+class Webhook(Hashable):
+    __slots__ = ( 'name', 'id', 'avatar', 'url', '_state',
+                 'webhook', 'token')
+    def __init__(self,state, **kwargs):
+        self._state = state
+        
+        self.name = kwargs.get('name',None)
+        self.id = kwargs.get('id',0)
+        self.token = kwargs.get('token','')
+        self.url = WEBHOOK_BASE + '{0.id}/{0.token}'.format(self)
+        self.avatar = AVATAR_BASE.format(id=self.id,
+                                         avatar=kwargs.get('avatar',''))
+        self.webhook = kwargs # Anything we didnt get will be in here
+    @classmethod
+    def _from_data(cls,state,data):
+        return cls(state,**data)
+    @asyncio.coroutine
+    def edit(self,*,update=False,**kwargs):
+        """
+        Set the webhooks information
+
+        Returns
+        ------
+            Updated discord.Webhook
+        """
+        if kwargs.get('name'): self.name = kwargs.get('name')
+        if kwargs.get('avatar'): self.avatar = kwargs.get('avatar')
+        if update: yield from self.update() # Updates the webhook
+        return self
+    @asyncio.coroutine
+    def update(self,**kwargs):
+        """|coro|
+        Updates the webhook
+        """
+        avatar = self.avatar
+        if self.avatar:
+            ret = yield from self._state.http._session.request('GET',self.avatar)

```

Comment:
Writing this should have triggered that something must have gone wrong in the implementation.